Community spotlight: Simplifying critical heart transplant waitlist data with Xata (#271)
xataio/mdx-blog: community-spotlight-heart-transplant-tracker.mdx

---
title: 'Simplifying critical heart transplant waitlist data with Xata'
description: "A community spotlight featuring Daniel Reale a QA engineer who used Xata to create an application for personally tracking his son's position on the Philadelphia heart transplant waitlist."
image:
  src: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/community-spotlight-educating-frontend-developers.jpg
  alt: Community Spotlight
author: Cezzaine Zaher
date: 06-13-2024
tags: ['community']
published: true
slug: community-spotlight-heart-transplant-tracker
---

> We're putting our spotlight on [Daniel Reale](https://www.linkedin.com/in/danreale/), a QA engineer by day at a leading data and advertising company, and a web application developer by night. His technical skills and personal dedication have driven him to create a vital application with a deeply personal mission - [a heart transport waitlist tracker](https://heart-transplant.netlify.app/).

Author: Awesome, thanks for reviewing and for spotting the typo!

> We're putting our spotlight on [Daniel Reale](https://www.linkedin.com/in/danreale/), a QA engineer by day at a leading data and advertising company, and a web application developer by night. His technical skills and personal dedication have driven him to create a vital application with a deeply personal mission - [a heart transport waitlist tracker](https://heart-transplant.netlify.app/).

Author: ```suggestion
We're putting our spotlight on [Daniel Reale](https://www.linkedin.com/in/danreale/), a QA engineer by day at a leading data and advertising company, and a web application developer by night. His technical skills and personal dedication have driven him to create a vital application with a deeply personal mission - [a heart transplant waitlist tracker](https://heart-transplant.netlify.app/).
```

---
title: 'Simplifying critical heart transplant waitlist data with Xata'
description: "A community spotlight featuring Daniel Reale a QA engineer who used Xata to create an application for personally tracking his son's position on the Philadelphia heart transplant waitlist."
image:
  src: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/heart-transplant-tracker/community-spotlight-heart-transplant-tracker.jpg
  alt: Community Spotlight
author: Cezzaine Zaher
date: 06-13-2024
tags: ['community']
published: true

> slug: community-spotlight-heart-transplant-tracker

Author: ```suggestion
slug: community-spotlight-heart-transplant-tracker
ogImage: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/heart-transplant-tracker/community-spotlight-heart-transplant-tracker.jpg
```

---

We're putting our spotlight on [Daniel Reale](https://www.linkedin.com/in/danreale/), a QA engineer by day at a leading data and advertising company, and a web application developer by night. His technical skills and personal dedication have driven him to create a vital application with a deeply personal mission - [a heart transplant waitlist tracker](https://heart-transplant.netlify.app/).

![Heart Transplant Tracker at a glance](/images/heart-transplant-tracker/tracker-at-a-glance-crop.png)

## The backstory

> Daniel's son, who recently turned five, is currently a patient at the Children's Hospital of Philadelphia, awaiting a heart transplant. Every day, Daniel and his wife anxiously monitor their son's position on the transplant list to stay informed and to communicate effectively with their transplant coordinator.

Author: ```suggestion
Daniel's son Colton, who recently turned five, is currently a patient at the Children's Hospital of Philadelphia, awaiting a heart transplant. Every day, Daniel and his wife anxiously monitor their son's position on the transplant list to stay informed and to communicate effectively with their transplant coordinator.
```

> ![Dan and his son](/images/heart-transplant-tracker/dan-son-500px.png)

Author: ```suggestion
![Dan and his son, Colton](/images/heart-transplant-tracker/dan-son-500px.png)
```

Daniel hopes that by sharing his story, it may help others going through similar experiences.
For more details on Daniel's project, you can visit the [Heart Transplant List Tracker app](https://heart-transplant.netlify.app/) and his Github repo at https://github.com/danreale/transplant. If you'd like to get in touch with Daniel, you can reach him on [LinkedIn](https://www.linkedin.com/in/danreale/).

## Feedback and feature requests

We asked Daniel what Xata features he found most useful and what he feels may be missing today. Here's what he shared:

- **Intuitive user interface:** the simplicity and visual ease that Xata provides because he prefers GUI-based interactions over scripting, and it's helped reduce the complexity around database management.
- **Flexible language queries:** the ability to use both Xata's domain-specific language and SQL for building queries has been very useful.

> I would definitely recommend Xata to other developers. It's been a fantastic tool for my side projects and has made a big difference in how I manage data.\
> _Daniel Reale - QA Engineer_

Areas for improvement include:

> - **Enhanced domain language features:** Daniel noted the need for more advanced functionalities for handling distinct records and limiting functions to avoid having to use standard SQL for certain tasks.

Reviewer: Not a deal breaker and I didn't notice this the last time I read through, but both of these areas for improvement almost mirror and conflict with what Daniel likes. If we were to address it, the quickest idea I had was to make "enhanced domain language features" and "usability issues" more specific to the actual problems.  So the first item could be "More helper functions in the SDK", and the second could be "More robust session handling" as examples.

Author: Thanks @alexfrancoeur, not sure if you saw this but it's here...

![Screenshot 2024-06-10 at 14 19 25](https://github.com/xataio/mdx-blog/assets/161334951/bf341599-542a-45ed-a4f1-af758983ea34)

> - **Enhanced domain language features:** Daniel noted the need for more advanced functionalities for handling distinct records and limiting functions to avoid having to use standard SQL for certain tasks.

Author: ```suggestion
- **Enhanced domain language features:** Daniel suggested more helper functions in the SDK as being something he'd find useful.
```

Author: @alexfrancoeur thoughts on this?

Reviewer: Looks great!

> - **Usability issues:** He experienced flashing issues with the table view after prolonged use, likely due to potential session handling or memory management problems, but we have not yet been able to replicate the issue.

Author: ```suggestion
- **Usability issues:** More robust session handling because he recently experienced flashing issues with the table view after prolonged use, likely due to potential session handling or memory management problems, but we have not yet been able to replicate the issue.
```

Author: @alexfrancoeur is this what you had in mind?

Reviewer: I was thinking of swapping "Usability issues" with "Robust session handling" or something along those lines. What do you think?

Author: Oh I see, sorry I misunderstood. Yes that works. I'll make the change.
